src/providers/TransactionBuilderContextProvider.tsx

```diff
@@ -781,6 +819,49 @@ const TransactionBuilderContextProvider = ({
     setShowTransactionBlockSelect(false);
   };
 
+  const connectionCheck = async () => {
+    if (sdk && connect) {
+      setIsWalletConnecting(true);
+      try {
+        const address = await connect();
+      } catch {}
+    }
+    setIsWalletConnecting(false);
+  };
+
+  useEffect(() => {
+    connectionCheck();
+  }, [sdk]);
+
+  useEffect(() => {
+    if (isWalletConnecting) {
+      setConnectionStatus(<Text color={theme.color?.text?.button}>Connecting</Text>);
+    } else if (accountAddress && !isWalletConnecting) {
+      setConnectionStatus(
+        <>
+          <StatusIconWrapper color={theme?.color?.background?.statusIconSuccess}>
+            <BiCheck size={12} />
+          </StatusIconWrapper>{' '}
+          <Text color={theme.color?.text?.button}>Connected</Text>
+        </>
+      );
+      setTimeout(() => {
+        setConnectionStatus(null);
+      }, 2000);
+    } else if (!accountAddress) {
+      setConnectionStatus(
+        <>
+          <StatusIconWrapper color={theme?.color?.background?.statusIconPending}>
+            <CgSandClock size={10} />
+          </StatusIconWrapper>
+          <Text color={theme.color?.text?.button}>Sign to connect</Text>
+        </>
+      );
+    } else {
+      setConnectionStatus(null);
+    }
```

Comment:
A switch case can be used here if possible instead of if else for a better readability.
```
const connectionStatus = useMemo(() => {
    switch (true) {
      case isWalletConnecting:
        return <Text color={theme.color?.text?.button}>Connecting</Text>;
      case accountAddress && !isWalletConnecting:
        return (
          <>
            <StatusIconWrapper color={theme?.color?.background?.statusIconSuccess}>
              <BiCheck size={12} />
            </StatusIconWrapper>
            <Text color={theme.color?.text?.button}>Connected</Text>
          </>
        );
      case !accountAddress:
        return (
          <>
            <StatusIconWrapper color={theme?.color?.background?.statusIconPending}>
              <CgSandClock size={10} />
            </StatusIconWrapper>
            <Text color={theme.color?.text?.button}>Sign to connect</Text>
          </>
        );
      default:
        return null;
    }
  }, [accountAddress, isWalletConnecting]);
```

Reply:
Yeah agreed